### PR TITLE
Feat/x chainspec types and tests configurator

### PR DESCRIPTION
**Pull request by meowsbits**
Fixing #68 

This change set establishes `interface`s for geth's understanding and using chain configuration.

These interfaces are then implemented in #70.

### CODE REVIEW

**Comment by meowsbits** [comment]
Removed `WIP` PR title prefix only because it is redundant to _Draft_ PR status.

File: build/env.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
```
Comment (tzdybal) [inline build/env.sh:1]: This file is obsolete because modules.

File: common/hexutil/json.go [hunk 2]
```diff
@@ -373,4 +380,4 @@ func wrapTypeError(err error, typ reflect.Type) error {
 
 func errNonString(typ reflect.Type) error {
 	return &json.UnmarshalTypeError{Value: "non-string", Type: typ}
-}
+}
```
Comment (tzdybal) [inline common/hexutil/json.go:383]: Different editor? ;)

**Review by tzdybal — APPROVED** [review]

**Comment by soc1c** [comment]
please resolve conflicts and check ci

File: common/hexutil/json.go (hunk 2 above)
Comment (meowsbits) [reply common/hexutil/json.go:383]: Yea... weird.

**Comment by meowsbits** [comment]
Closing in favor of #82. 

For resolution of review comment https://github.com/etclabscore/multi-geth/pull/69#discussion_r355335147, see #83

**Pull request closed without merging**